text/0517-io-os-reform.md

```diff
@@ -483,7 +483,72 @@ throughout IO, we can go on to explore the modules in detail.
 ### `std::env`
 [std::env]: #stdenv
 
-> To be added in a follow-up PR.
+Most of what's available in `std::os` today will move to `std::env`,
+and the signatures will be updated to follow this RFC's
+[Design principles] as follows.
+
+**Arguments**:
+
+* `args`: change to yield an iterator rather than vector if possible; in any case, it should produce an `OsStrBuf`.
```

Comment:
I'm sorry I'm not really understanding where your questions are going, do you have a particular implementation in mind that you would like to see? If you've got comments about a specific implementation, it may also be helpful to reference https://github.com/alexcrichton/io2 which has this API implemented.

> How does the second sentence follow from the first? If you get argc/v from main then the only thing you have to store are argc/v, not the memory behind arcv.

I don't think I understand the question here. We make a copy of argv to store ourselves, but we may not need to (haven't investigated very thoroughly).